### ISSUE TITLE
Incorrect parsing of unknown date

### ISSUE THREAD

**Issue by barnaclebarnes** [inukshuk/edtf-ruby#16]
A completely unknown date can be written as `uuuu` however this fails to parse:

```
➜  pncc-data git:(master) ✗ irb
irb(main):001:0> require 'edtf'
=> true
irb(main):002:0> EDTF.parse!('uuuu')
ArgumentError: failed to parse date: unexpected 'u' at []
	from parser.y:473:in `on_error'
	from /Users/barnacle/.rbenv/versions/2.2.4/lib/ruby/2.2.0/racc/parser.rb:258:in `_racc_do_parse_c'
	from /Users/barnacle/.rbenv/versions/2.2.4/lib/ruby/2.2.0/racc/parser.rb:258:in `do_parse'
	from parser.y:469:in `parse!'
	from /Users/barnacle/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/edtf-2.2.0/lib/edtf/date.rb:30:in `edtf!'
	from /Users/barnacle/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/edtf-2.2.0/lib/edtf.rb:83:in `parse!'
	from (irb):2
	from /Users/barnacle/.rbenv/versions/2.2.4/bin/irb:11:in `<main>'
```

**Comment by inukshuk** [comment]
True. Parsing this is easy, but it's not immediately clear what sort of value to expect as a result. I'd propose to use a special value, akin to Infinity for Number; this value must act like a date (for comparisons etc.). Any thoughts?

**Comment by barnaclebarnes** [comment]
What we are trying to do is to distinguish between `nil` dates and `unknown` dates so a special value would work for us. I'm not sure what `min` and `max` should return if anything.

**Comment by inukshuk** [comment]
I just pushed a very quick fix for this. Parsing 'uuuu' will return an instance of the new `EDTF::Unknown` which so far doesn't do much but inherit from `Date::Infinity`. As a next step we should consider how this object should compare to other dates, intervals, season etc.

Anyway, let me know what you think; if it is sufficient for you as is I'm tempted to just release it for now. Once the ISO adoption process is more stable the parse will have to be adjusted and I was not planning to put any effort into the Gem until then :)

**Comment by barnaclebarnes** [comment]
That works for me and is sufficient for now. I agree, waiting until the ISO adoption process completes before putting much extra effort is the way to go.

**Comment by inukshuk** [comment]
Landed in 2.3.0
